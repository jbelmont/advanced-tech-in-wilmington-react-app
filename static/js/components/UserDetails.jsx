import React, { Component } from 'react';
import {browserHistory} from 'react-router';

import store from '../store/store';

const UserDetails = routerInfo => {
    const userInformation = store && store.getState() && store.getState()["userInfo"];
    let email, firstName, lastName, gender, id;
    if (Object.keys(userInformation).length > 0) {
        email = userInformation["email"];
        firstName = userInformation["firstName"];
        lastName = userInformation["lastName"];
        gender = userInformation["gender"];
        id = userInformation["id"];
    } else {
        const storeInfo = store && store.getState() && store.getState()["users"];
        firstName = storeInfo["first_name"];
        lastName = storeInfo["last_name"];
        email = storeInfo["email"];
        gender = storeInfo["gender"];
        id = storeInfo["id"];
    }

    const UserDetailsArea = (
        <div className="user-details-container">
            <span className="email">{email}</span>
            <span className="first-name">{firstName}</span>
            <span className="last-name">{lastName}</span>
            <span className="gender">{gender}</span>
        </div>
    );
    return (
        <div className="triangle-react-container">
        <h2 className="triangle-react-container-label"><strong>Triangle ReactJS Users</strong></h2>
        {UserDetailsArea}
      </div>
    );
}

export default UserDetails;
