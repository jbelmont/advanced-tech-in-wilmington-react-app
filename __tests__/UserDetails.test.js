'use strict';

jest.mock('../static/js/data/index', () => 'data');
const mock = jest.fn();
mock('store');
jest.mock('../static/js/constants', () => 'constants');

import React from 'react';
import UserDetails from '../static/js/components/UserDetails';
import renderer from 'react-test-renderer';

test('Test the <UserDetails /> component', () => {
  const component = renderer.create(
    <UserDetails
      email='johnrambo@badass.net'
      firstName='John'
      lastName='Rambo'
      gender='Male'
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
