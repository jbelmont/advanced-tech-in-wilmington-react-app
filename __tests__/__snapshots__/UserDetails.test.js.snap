exports[`test Test the <UserDetails /> component 1`] = `
<div
  className="advanced-tech-container">
  <h2
    className="advanced-tech-container-label">
    <strong />
  </h2>
  <div
    className="user__details-container">
    <span
      className="user__details-container-email" />
    <span
      className="user__details-container-first-name" />
    <span
      className="user__details-container-last-name" />
    <span
      className="user__details-container-gender" />
  </div>
</div>
`;
